### PR TITLE
[k8s] Refactor CrdWatcher

### PR DESCRIPTION
**Pull request by dmolokanov**
- Renamed CrdWatchOperator to EdgeDeploymentOperator
- Renamed CrdWatcher to EdgeDeploymentController
- EdgeDeploymentOperator only 
    1. watch CRD and delegates modules deployment on k8s to EdgeDeploymentController
    2. stores current modules
- EdgeDeploymentController make all checks against k8s api and deploy modules with services and service accounts

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                            await this.HandleEdgeDeploymentsChangedAsync(type, item);
+                        }
+                        catch (Exception ex) when (!ex.IsFatal())
+                        {
+                            Events.ExceptionInCustomResourceWatch(ex);
+                        }
+                    },
+                    onClosed: () =>
+                    {
+                        Events.CrdWatchClosed();
```
Comment (darobs) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs]: A really minor comment: If we've renamed everything else, we should also update the Events name...

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                ExceptionInCustomResourceWatch = IdStart,
+                DeploymentStatus,
+                DeploymentError,
+                DeploymentNameMismatch,
+                CrdWatchClosed,
+            }
+
+            public static void ExceptionInCustomResourceWatch(Exception ex)
+            {
+                Log.LogError((int)EventIds.ExceptionInCustomResourceWatch, ex, "Exception caught in Custom Resource Watch task.");
```
Comment (darobs) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs]: ...and the debug messages.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs (hunk 1 above)
Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs]: Good. Will do it in the next PR

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs (hunk 2 above)
Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeDeploymentOperator.cs]: sure

**Pull request merged**